includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -73,23 +73,40 @@ class AMP_Gallery_Block_Sanitizer extends AMP_Base_Sanitizer {
 	 * @since 0.2
 	 */
 	public function sanitize() {
-		$nodes     = $this->dom->getElementsByTagName( self::$tag );
-		$num_nodes = $nodes->length;
-		if ( 0 === $num_nodes ) {
-			return;
-		}
+		$xpath = new DOMXPath( $this->dom );
+		$expr  = sprintf(
+			'//ul[ %s ]',
+			implode(
+				' or ',
+				[
+					'( parent::figure[ contains( @class, "wp-block-gallery" ) and @data-amp-carousel = "true" ] )',
+					'( contains( @class, "wp-block-gallery" ) and @data-amp-carousel = "true" )',
```

Comment:
I believe you've lost some functionality with this change. Previously you could have the carousel "off" and lightbox "on" for the gallery, but this sanitizer will now ignore that setting combination.